### PR TITLE
qq: Add version 9.7.3.28946

### LINKED ISSUE #10558: [Request]: QQ
### Prerequisites

- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.

### Criteria

- [X] Reasonably well-known and widely used (e.g. if it's a GitHub project, it should have at least 100 stars and/or 50 forks)
- [X] English interface (or at least English documentation)
- [X] Latest stable version
- [X] Full version (i.e. not a trial version)
- [X] Fairly standard install (e.g. uses a version-specific download URL, no elaborate pre/post install scripts)

### Name

qq

### Description

An instant messaging tool that gives you the best way to keep in touch with your friends and family.

### Homepage

https://im.qq.com/pcqq

### Download Link(s)

https://dldir1.qq.com/qqfile/qq/PCQQ9.7.3/QQ9.7.3.28946.exe

### Some Indication of Popularity/Repute

QQ is an instant messaging software that is widely used in mainland China and is also used by some foreigners when they study and work in China. The software is produced by Tencent and has a very large number of users.

### PR DESCRIPTION
**Pull request by StarsbySea**
Closes #10558 
- [X] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by niheaven** [comment]
Do QQ and TIM have English UI after installed? Since I'm using Simplified Chinese locale and could not test them.

**Comment by StarsbySea** [comment]
I hope this gets merged, but I'm also not going to lie, the answer is No and No.
However after testing the system is in English, the software can display Simplified Chinese and work fine. Of course as a chat software, even if the interface is all in Simplified Chinese it still works (just chatting)

**Comment by niheaven** [comment]
So, these manifests should go to other buckets IMO, e.g. @chawyehsu's https://github.com/chawyehsu/dorado, and there're many other Chinese only ones.

**Comment by chawyehsu** [comment]
My review for it: objection.
Not only because of the language criteria. I do not think this will work by simply extracting files from the installers, neither QQ nor TIM. You have to investigate and ensure it really works in a new environment that did not have QQ/TIM, more specifically `QQProtect`, installed before. Otherwise: https://github.com/chawyehsu/dorado/issues/11, https://github.com/chawyehsu/dorado/issues/370.

**Comment by StarsbySea** [comment]
> So, these manifests should go to other buckets IMO, e.g. @chawyehsu's https://github.com/chawyehsu/dorado, and there're many other Chinese only ones.

I am actually aware of @chawyehsu's https://github.com/chawyehsu/dorado, but I still want to submit that inventory file to the Extras repository for several reasons.
1. Extras has a reputation as one of the few official repositories, maintained by a wide range of professional community members, and it would be convenient for many users who just want to install software to be added to it. (rather than finding an uncertain third party repository to run potentially risky scripts or software)
2. Searching through the Scoop.sh website, you can find at least 20+ repositories that provide QQ manifests, suggesting that many Scoop users actually use the software's configuration, perhaps because the official Extras don't have it and have to (painfully) create it themselves
3. In the face of objections from https://github.com/ScoopInstaller/Main/pull/2938#issuecomment-966830432, I would like to say that if a software has a large enough user base and if the users need it, it is possible to consider Allowing software to be added to a repository. Maintenance is not really a problem, because if the software is used by enough people and is open source, there will be users who will raise issues, or pull requests to fix them.
Of course, in the specific case of QQ and TIM, the software has a simplified Chinese interface, and there are at least two maintainers I know of who can read Chinese, so I don't think it should be too difficult to maintain. (All of this is debatable, of course).

**Comment by StarsbySea** [comment]
> My review for it: objection. Not only because of the language criteria. I do not think this will work by simply extracting files from the installers, neither QQ nor TIM. You have to investigate and ensure it really works in a new environment that did not have QQ/TIM, more specifically `QQProtect`, installed before. Otherwise: [chawyehsu/dorado#11](https://github.com/chawyehsu/dorado/issues/11), [chawyehsu/dorado#370](https://github.com/chawyehsu/dorado/issues/370).

Thank you for your comments.

I guess I didn't post a screenshot though, or you didn't actually go and test it to conclude that it wouldn't work without QQ Protect.
**To be honest, I've tested both manifest files in Windows Sandbox (must admit the sandbox must be clean Windows. test was done by copying the finished QQ/TIM folder into the sandbox, running it manually and trying to log in) and both worked fine with no error messages.**
I've posted screenshots of the test below for you to see, but a better way is to try it out for yourself. (Just a personal opinion, it would be a waste of time and emotion and not good for community dialogue and development if you just thought about it and didn't even bother to try it out before coming to a conclusion)

QQ
TIM

**Comment by chawyehsu** [comment]
Claiming third-party buckets are risky is somewhat negative to the community. Most of them are created by community members because of their own pure needs. But do get me wrong, ppl should always double-check the packages they use.

> ..  @ chawyehsu's https://github.com/chawyehsu/dorado, and there're many ..

My bucket was not for such a situation from the beginning. I created it for myself and built up its reputation on my own. No one else has a duty to submit any manifest that does not satisfy Scoop org-owed buckets' criteria to my bucket. **So please do not give such recommendations.** We, at least I and @ rasa, expect there should be more buckets that are built and maintained by the community members themselves, and stuff related to buckets works in a more _decentralized_ way. Remember you can create your own bucket. (Ross tell me if I'm wrong)

In fact, I don't quite care about the language-specific maintenance criteria after `winget` started to include almost all incoming packages without basically any clear requirements. I don't remember who introduced and enforced such criteria for our buckets. I do care about package usability, especially for restricted users whom Scoop was made for, naming convention, and maintainability, however. That's another story. Not going to discuss it here. On the other hand, I'll be excited if anyone makes these two apps work with Scoop without requiring admin rights. I'll be happy to adapt then. Anyways I'll apologize for my preconceived comment and have it a try under VMs using a restricted user account. I'll review the manifest and approve this when it works under my VM, but if there is any objection from other collaborators, it's up to them.

**Comment by niheaven** [comment]
Okay, I'll not recommend your bucket since it's *personal* as you said.

I've checked git log of `package_request` issue template and the language criteria is introduced by me but I'm sure it's not from me, and also forget who said this first. If there's a need to accept all types of apps, we could remove any of these criteria and make official buckets more general ones.

But there're over 1800 apps in `extras` with over 100 package-adding PRs and the sub-directory structure (in any categories) is still not adopted 'cos incompatibility of search-engine, I'm really worried about the maintainability of this bucket.

File: bucket/qq.json
Comment (chawyehsu) [inline bucket/qq.json:32]: This is not a good practice as it'll erase the runtime-generated data every time an upgrade is triggered. And these data will be re-generated when the app starts. Hence it makes no sense. App data should only be manipulated while the `--purge` flag is used.

Comment (StarsbySea) [reply bucket/qq.json]: I have modified it according to your comment and added some persist configuration, please check it out and feel free to review and comment.

File: bucket/qq.json
Comment (chawyehsu) [reply bucket/qq.json:51]: Thoughts? /cc @ScoopInstaller/maintainers

**Comment by chawyehsu** [comment]
> But there're over 1800 apps in extras with over 100 package-adding PRs and the sub-directory structure (in any categories) is still not adopted 'cos incompatibility of search-engine, I'm really worried about the maintainability of this bucket.

I'm concerned about this, too. Still, you have to admit that ppl thinking of having a manifest included in org-owned buckets that would leverage the reputation will keep submitting new apps whatever criteria are there, they can actually create their own buckets as I comment above tho. We have to find a solution to solve this rather than add limitations. https://github.com/ScoopInstaller/Scoop/pull/5119 was great enough though.

File: bucket/qq.json
Comment (rashil2000) [reply bucket/qq.json:51]: From the `post_install` I can see we're trying to _force_ persistence here. I do not advocate this: it can lead to all kinds of unexpected behaviour. If the app stores data in the Documents or the AppData folder, we should let it. 

Only if an application is _by design_ portable, we should manipulate its data.

Comment (StarsbySea) [reply bucket/qq.json]: You're right!

File: bucket/qq.json
Comment (niheaven) [inline bucket/qq.json:38]:
```suggestion
    "installer": {
        "script": [
            "Get-Item \"$dir\\*\" -Exclude .rsrc | Remove-Item",
            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\101\" \"$dir\" -ExtractDir 'Files'",
            "Expand-7zipArchive \"$dir\\.rsrc\\2052\\MSI\\102\" \"$dir\\temp\"",
            "'vcruntime140.dll', 'msvcp140.dll', 'msvcr100.dll', 'vcomp140.dll', 'msvcp100.dll' | ForEach-Object {",
            "  Move-Item \"$dir\\temp\\$_\" \"$dir\\Bin\\\"",
            "}",
            "Expand-7zipArchive \"$dir\\temp\\TXSSO\\TXSSOSetup.exe\" \"$dir\\Bin\\TXSSO\" -ExtractDir 'TXSSO'",
            "Remove-Item \"$dir\\.rsrc\", \"$dir\\temp\" -Recurse"
        ]
    },
```

Simplified `installer.script`, and got same folder structure. Please test it.

image

File: bucket/qq.json
Comment (niheaven) [inline bucket/qq.json:40]:
```suggestion
        "url": "https://pc.qq.com/detail/2/detail_2.html",
```

**Review by niheaven — COMMENTED** [review]
Simplify script and change checkver url

**Comment by StarsbySea** [comment]
> Simplify script and change checkver url

Thank you. Local test passed.

**Review by rashil2000 — APPROVED** [review]

**Pull request merged**
